Calendly: Changed the embed, to remove the inline styles

pablinos opened the pull request:
The inline styles added to the parent container by the Calendly script
were interfering with the CSS used for aligning elements to the left
and right on the desktop view of Twenty Twenty (and so probably other
themes)

#### Changes proposed in this Pull Request:

An undocumented option to the init function stops these styles being
added, and so this change swaps the embed code to set that option, as
well as tweak the CSS to make the best effort at laying out the embed when
aligned to the sides.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
This an improvement on a new feature

#### Testing instructions:

* Embed a Calendly block and set the alignment to left or right
* Add some lorem ipsum text to the post
* Preview the post at different screen sizes. (You'll have to reload the page sometimes as the Calendly code sets the width of the embed on load and doesn't always resize it)
* Check that the embed isn't (too) broken. Previously the embed would disappear over screens 1000px wide and be partially cut off by the side of the screen. 

There are still some known problems with the Twenty Twenty theme and alignment of blocks, but having discussed it with @kjellr (p1580823343135400-slack-themes), we've decided that it's best to leave the layout to the theme and make the block as lightweight as possible.

#### Proposed changelog entry for your changes:
* No changelog required

pablinos committed:
Changed the embed, to remove the inline styles

The inline styles added to the parent container by the Calendly script
were interferring with the CSS used for aligning elements to the left
and right on the desktop view of Twenty Twenty (and so probably other
themes)

An undocumented option to the init function stops these styles being
added, and so this change swaps the embed code to set that option, as
well as tweak the CSS to make a best effort at laying out the embed when
aligned to the sides.

pablinos committed:
Changed the embed, to remove the inline styles

The inline styles added to the parent container by the Calendly script
were interferring with the CSS used for aligning elements to the left
and right on the desktop view of Twenty Twenty (and so probably other
themes)

An undocumented option to the init function stops these styles being
added, and so this change swaps the embed code to set that option, as
well as tweak the CSS to make a best effort at laying out the embed when
aligned to the sides.

pablinos committed:
Scoped the height setting to the inline style

pablinos committed:
Scoped the height setting to the inline style

pablinos committed:
Prevent the server side rendering in the admin

The render_callback was being called to get the content of the page in
the admin, and this was causing the block embed to render full screen
when it was inline mode. This short circuits the render callback in the
admin.

pablinos committed:
Prevent the server side rendering in the admin

The render_callback was being called to get the content of the page in
the admin, and this was causing the block embed to render full screen
when it was inline mode. This short circuits the render callback in the
admin.

scruffian commented:
This looks good to me, just two thoughts:

1. If this feature is undocumented is it a good idea to depend on it?

2. As you said, when you resize the window the layout gets stuck:
<img width="1440" alt="Screenshot 2020-02-10 at 12 58 12" src="https://user-images.githubusercontent.com/275961/74152023-57372280-4c05-11ea-888a-a171ef020295.png">

I wonder if we can fire an event on resize to sort this out? I added an issue for it: https://github.com/Automattic/jetpack/issues/14619

pablinos commented:
> * If this feature is undocumented is it a good idea to depend on it?

Well, no, but there's no other option really. If we allow it to add the inline styles it breaks things, and to override it, we'd probably have to get into a lot of theme-specific CSS, or sprinkling `!important` around, which doesn't seem like a good idea either.

> * As you said, when you resize the window the layout gets stuck:

Yes, I don't think there's much we can do about that at the moment. It's all wrapped up with the problem of alignment and the way these aligned embeds are positioned using `absolute`.

pablinos committed:
Simplify the block ID code, and remove is_admin check

We've discovered using `wp_unique_id`, so it seemed sensible to use that
here.

There was a check for is_admin in the render callback, because of an
error with the post being rendered as the editor loaded. I can't
replicate this now, so I've removed the check.

scruffian committed:
fix rebase

jeherve commented on extensions/blocks/calendly/edit.js:
```diff
@@ -250,8 +250,10 @@ export default function CalendlyEdit( { attributes, className, clientId, setAttr
 		</InspectorControls>
 	);
 
+	const classes = `${ className } calendly-style-${ style }`;
```
You may have to switch to `getBlockDefaultClassName` like you did in #14725?

pablinos replied:
I think `className` is good here as that means we'll have any additional classes that have been added in the advanced settings, and the `calendly-style-` class won't be affected by that issue.

jeherve commented on extensions/blocks/calendly/calendly.php:
```diff
@@ -148,10 +153,18 @@ function load_assets( $attr, $content ) {
+	inlineStyles: false,
+});
+JS_END;
+		wp_add_inline_script( 'jetpack-calendly-external-js', sprintf( $script, esc_url( $url ), $block_id ) );
```
I would `esc_js` the unique ID here just to be on the safe side.

scruffian replied:
Done

jeherve commented on extensions/blocks/calendly/calendly.php:
```diff
@@ -148,10 +153,18 @@ function load_assets( $attr, $content ) {
+			'<div class="%1$s" id="%2$s"></div>',
 			esc_attr( $classes ),
-			esc_url( $url )
+			$block_id
```
Let's escape this as well.

scruffian replied:
Done

jeherve commented on extensions/blocks/calendly/calendly.php:
```diff
@@ -79,6 +79,11 @@ function set_availability() {
+	if ( is_admin() ) {
+		return;
+	}
+	static $block_num = 0;
```
Do we still need this?

scruffian replied:
No, removed

jeherve reviewed (changes requested):
This works for me, I only have minor comments left.

scruffian committed:
properly escape the block_id

scruffian commented:
Ready for another review.

jeherve reviewed (approved):
This works well in my tests. 👍

pablinos commented:
r203460-wpcom
